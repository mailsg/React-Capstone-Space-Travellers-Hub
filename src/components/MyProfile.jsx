import { useSelector } from 'react-redux';
import '../App.css';

const MyProfile = () => {
  const { rockets } = useSelector((store) => store.rockets);
  const reserveRocket = rockets.filter((rocket) => rocket.reserved);

  const missionsList = useSelector((state) => state.missions.missions);
  const selectedMission = missionsList.filter((mission) => mission.reserved);

  return (
    <div className="profile mx-5">
      <div className="my-container">
        <h2>My Missions</h2>
        <ul>
          {selectedMission.map((mission) => (
            <li key={mission.mission_id} className="fw-bold">
              {mission.mission_name}
            </li>
          ))}
        </ul>
      </div>
      <div className="my-container">
        <h2>My Rockets</h2>
        <ul>
          {reserveRocket.map((rocket) => (
            <li key={rocket.id} className="fw-bold">
              {rocket.name}
            </li>
          ))}
        </ul>
      </div>
    </div>
  );
};

export default MyProfile;
