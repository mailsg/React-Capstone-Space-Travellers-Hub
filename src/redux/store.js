import { configureStore } from '@reduxjs/toolkit';
import rocketsSlice from './rockets/rocketSlice';
import missionsSlice from './missions/missionsSlice';

const store = configureStore({
  reducer: {
    rockets: rocketsSlice,
    missions: missionsSlice,
  },
});

export default store;
