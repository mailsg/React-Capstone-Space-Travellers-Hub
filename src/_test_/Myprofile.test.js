import '@testing-library/jest-dom';
import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import React from 'react';
import store from '../redux/store';
import MyProfile from '../components/MyProfile';

describe('User Interactions test', () => {
  it('will test user interaction', () => {
    const tree = renderer
      .create(
        <Provider store={store}>
          <MyProfile />
        </Provider>,
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
