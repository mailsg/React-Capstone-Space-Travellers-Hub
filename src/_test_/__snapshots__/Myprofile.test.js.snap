// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`User Interactions test will test user interaction 1`] = `
<div
  className="profile mx-5"
>
  <div
    className="my-container"
  >
    <h2>
      My Missions
    </h2>
    <ul />
  </div>
  <div
    className="my-container"
  >
    <h2>
      My Rockets
    </h2>
    <ul />
  </div>
</div>
`;
